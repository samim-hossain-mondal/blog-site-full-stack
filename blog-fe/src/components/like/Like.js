import React, { useState } from 'react';
import heartBlack from '../../assets/icons/heart-black.svg';
import heartRed from '../../assets/icons/heart-red.svg';

export default function Like() {
  const [like, setLike] = useState(0);
  const likeHandler = () => {
    if (like === 0) {
      setLike(like + 1);
    } else {
      setLike(like - 1);
    }
  };
  return (
    <div className='like'>
      <img className='icons' data-testid="like-btn" onClick={likeHandler} src={like ? heartRed : heartBlack} />
    </div>
  );
}