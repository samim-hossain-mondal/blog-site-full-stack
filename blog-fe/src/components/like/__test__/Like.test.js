/* eslint-disable no-undef */
import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Like from '../Like';

describe('Like', () => {
  it('should render', () => {
    const { asFragment } = render(<Like />);
    expect(asFragment()).toMatchSnapshot();
  });
  describe('when the user clicks on the like button', () => {
    it('should change the color of the like button', () => {
      const { queryByTestId } = render(<Like />);
      const likeBtn = queryByTestId('like-btn');
      fireEvent.click(likeBtn);
      expect(likeBtn).toHaveAttribute('src', 'heart-red.svg');
    });
  });
});