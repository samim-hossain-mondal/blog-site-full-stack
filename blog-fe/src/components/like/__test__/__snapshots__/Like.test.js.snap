// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Like should render 1`] = `
<DocumentFragment>
  <div
    class="like"
  >
    <img
      class="icons"
      data-testid="like-btn"
      src="heart-black.svg"
    />
  </div>
</DocumentFragment>
`;
